runtime/src/lib.rs

```diff
@@ -62,10 +62,10 @@ pub enum Error {
     BindFailed,
     #[error("connection failed")]
     ConnectionFailed,
-    #[error("write failed")]
-    WriteFailed,
-    #[error("read failed")]
-    ReadFailed,
+    #[error("write failed. error: {0}")]
+    WriteFailed(IoError),
```

Comment:
May want to go to this: https://github.com/commonwarexyz/monorepo/blob/main/codec/src/error.rs#L55-L73